homeassistant/components/glances/sensor.py

```diff
@@ -19,33 +19,79 @@
 async def async_setup_entry(hass, config_entry, async_add_entities):
     """Set up the Glances sensors."""
 
-    glances_data = hass.data[DOMAIN][config_entry.entry_id]
+    client = hass.data[DOMAIN][config_entry.entry_id]
     name = config_entry.data[CONF_NAME]
     dev = []
-    for sensor_type in SENSOR_TYPES:
-        dev.append(
-            GlancesSensor(glances_data, name, SENSOR_TYPES[sensor_type][0], sensor_type)
-        )
+
+    for sensor_type, sensor_details in SENSOR_TYPES.items():
+        if sensor_details[0] in client.api.data:
+            if sensor_details[0] == "fs":
+                # fs will provide a list of disks attached
+                for disk in client.api.data[sensor_details[0]]:
+                    dev.append(
+                        GlancesSensor(
+                            client,
+                            name,
+                            disk["mnt_point"],
```

Comment:
Can we somehow also put this in `SENSOR_TYPES` ? Maybe a lambda with extract info: `lambda item: item["mnt_point"]`

Reply:
I need to keep it as perfix because i use the prevdix value when updating the entity in `async_update`